### PR TITLE
Backend/feature/sign in

### PR DESCRIPTION
**Pull request by pinola-b**

### CODE REVIEW

**Comment by Lank891** [comment]
Just reminder about assignee, label, project and milestone :D

**Comment by pinola-b** [comment]
I made a separate table for Shelters, and made a OneToOne connection between DogShelterAccount and Shelter.
Had to create an additional wrapper class for LoginInformation. 

Endpoints for authentication weren't specified (or I am just blind), so I did it as /account/login

**Comment by pinola-b** [comment]
Would close #13 (Sign-in in backend)

File: Backend/src/test/java/controller/internal/UserControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public void authenticateTest() throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
+        String json = objectMapper.writeValueAsString(new LoginInfo("e.musk@mail.com", "xea-12Musk"));
+        System.out.println(json);
+        mockMvc.perform(
+                MockMvcRequestBuilders.post("/account/login")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(json)
+                        .characterEncoding("utf-8"))
+                .andExpect(status().isOk());
```
Comment (Lank891) [inline Backend/src/test/java/controller/internal/UserControllerTest.java:39]: This has to be changed obviously after api changes, check not only if it is ok but for example if user type is correct

**Review by Lank891 — CHANGES_REQUESTED** [review]
It is nice, you can make login request and you get the token. But the token is not stored anywhere. You have to store the token (alongside with the permission type like admin/shelter/user). You also need a function (yeah it is not in the spec but it is needed as internal implementation) that will take the token (or whole http header) and check, if token is actually valid and the permission matches (for example it can be a function in login service `boolean IsAuthorized(string token,  UserType requiredPermissions)` (or with headers instead of token, like in carly). Additionally it would be nice (if you won't do it, i'll do it as a separate PR) to add exception for lack of authorisation, handle it to return code 403 (according to api docs) and add it already to existing getting and adding dogs. Obviously then tests must be changed since right now they do not have token and we would need to create some kind of test token for development anyway.

**Review by Lank891 — APPROVED** [review]
Todo: 
1. Add tests with failed login
2. Add requests authentication

I'll do it in a separate PR just as I said before

**Pull request merged**